docs/DeployConfigurations.rst

```diff
@@ -55,7 +63,37 @@ The simplest configuration is when Nextcloud is installed on the host and Docker
 		class ExApp2 python
 		class ExApp3 python
 
-Suggested way to communicate with Docker: via ``socket``.
+Suggested way to communicate with Docker: via ``docker-socket-proxy``.
+
+.. mermaid::
+
+	stateDiagram-v2
+		classDef docker fill: #1f97ee, color: transparent, font-size: 34px, stroke: #364c53, stroke-width: 1px, background: url(https://raw.githubusercontent.com/cloud-py-api/app_api/main/docs/img/docker.png) no-repeat center center / contain
+		classDef nextcloud fill: #006aa3, color: transparent, font-size: 34px, stroke: #045987, stroke-width: 1px, background: url(https://raw.githubusercontent.com/cloud-py-api/app_api/main/docs/img/nextcloud.svg) no-repeat center center / contain
+		classDef python fill: #1e415f, color: white, stroke: #364c53, stroke-width: 1px
+
+		Host
+
+		state Host {
+			Nextcloud --> DockerSocketProxy: by port
+			Docker --> Containers
+			Docker --> DockerSocketProxy : /var/run/docker.sock
+
+			state Containers {
+				DockerSocketProxy --> ExApp1
+				DockerSocketProxy --> ExApp2
+				DockerSocketProxy --> ExApp3
+			}
+		}
+
+		class Nextcloud nextcloud
+		class Docker docker
+		class ExApp1 python
+		class ExApp2 python
+		class ExApp3 python
+
+
+`Docker Socket Proxy container <https://github.com/nextcloud/all-in-one/tree/main/Containers/docker-socket-proxy>`_ can be used for that.
```

Comment:
Should we maybe add some more details how to use that? Also I think we would need to add some changes to the container to make it work in more environments?

Reply:
I think, its a good idea and it would be much easier and more convenient for admins.
Exposing Docker Engine API with TLS using Docker container with our strictly configured rules sounds good.

Reply:
So should we start modifying the container in order to allow to use it in more environments?

Reply:
Yes, we already displaying it on schemes, so we can't merge this PR without it.

Reply:
@szaimen Hey! What is needed from us regarding modifying docker-socket-proxy container? It probably will not that much different, we just need to document required ENVs for it and how to run it.

Reply:
> we just need to document required ENVs for it and how to run it.

Yes indeed, however I am currently very busy with other stuff so a PR that adds these ENV to the dockerfile of the container is very welcome!